src/effects/How to make an effect stateless.txt

```diff
@@ -0,0 +1,189 @@
+    
+    add line "struct Validator;" in public section
+
+
+    In cpp file:
+    add class definition in the .cpp with mSettings:
+    
+        struct EffectEcho::Validator
+           : DefaultEffectUIValidator
+        {
+           Validator(EffectUIClientInterface& effect,
+              EffectSettingsAccess& access, EffectEchoSettings& settings)
+              : DefaultEffectUIValidator{ effect, access }
+              , mSettings{ settings }
+           {}
+           virtual ~Validator() = default;
+
+           Effect& GetEffect() const { return static_cast<Effect&>(mEffect); }
+
+           bool ValidateUI() override;
+           bool UpdateUI() override;
+
+           void PopulateOrExchange(ShuttleGui& S);
+
+           EffectEchoSettings& mSettings;
+        };
+
+
+        bool EffectEcho::Validator::ValidateUI()
+        {
+           mAccess.ModifySettings
+           (
+              [this](EffectSettings& settings)
+           {
+              // pass back the modified settings to the MessageBuffer
+
+              // TODO uncomment at last step
+              //EffectEcho::GetSettings(settings) = mSettings;
+           }
+           );
+
+           return true;
+        }
+
+
+        bool EffectEcho::Validator::UpdateUI()
+        {
+           // get the settings from the MessageBuffer and write them to our local copy
+           const auto& settings = mAccess.Get();
+
+           // TODO uncomment at last step
+           //mSettings = GetSettings(settings);
+
+           return true;
+        }
+
+    
+
+
+3) "Define the instance"
+
+    In header file, 
+    
+        add in public section:
+    
+            struct Instance;
+
+            std::shared_ptr<EffectInstance> MakeInstance(EffectSettings& settings) const;
+
+        Identify members that represent state - comment them out 
+        
+        remove methods: ProcessInitialize, ProcessFinalize, ProcessBlock
+
+    In cpp file,
+    
+        add:
+    
+        struct EffectEcho::Instance
+           : public PerTrackEffect::Instance
+           , public EffectInstanceWithBlockSize
+           , public EffectInstanceWithSampleRate
+
+        {
+           Instance(const PerTrackEffect& effect)
+              : PerTrackEffect::Instance{ effect }
+           {}
+
+           bool ProcessInitialize(EffectSettings& settings,
+              sampleCount totalLen, ChannelNames chanMap) override;
+
+           size_t ProcessBlock(EffectSettings& settings,
+              const float* const* inBlock, float* const* outBlock, size_t blockLen)  override;
+
+           bool ProcessFinalize(void) override;
+
+           << MEMBERS WHICH REPRESENT STATE which you commented out earlier >>
+        };
+
+
+        std::shared_ptr<EffectInstance>
+        EffectEcho::MakeInstance(EffectSettings&) const
+        {
+           return std::make_shared<Instance>(*this);
+        }
+        
+        
+    Make the 3 Effect::Process*** methods belong to the instance by: 
```

Comment:
For some effects, this will be more than 3.  See the example of LadspaEffect::Instance in the waiting PR.